sklearn/cluster/_spectral.py

```diff
@@ -191,6 +193,20 @@ def discretize(
     return labels
 
 
+
+@validate_params(
+    { 
+        "affinity": ["array-like", "sparse matrix"],
+        "n_clusters": [Interval(Integral, 1, None), None],
+        "n_components": [Interval(Integral, 1, None), None],
+        "eigen_solver":[StrOptions({'arpack', 'lobpcg', 'amg', None})],
+        "random_state":[Interval(Integral, 1, None), None],
+        "n_init": [Interval(Integral, 1, None), None],
+        "eigen_tol": [Interval(Real, 0.0, None,), None],
+        "assign_labels":[StrOptions({'kmeans', 'discretize', 'cluster_qr'})],
+        "verbose": ["boolean"],
+    }
+)
```

Comment:
We will have to make a refactoring first to create a `SpectralClustering` instance in the function as done in previous PR.